apps/api-key-service/src/models/apiKeys.ts

```diff
@@ -73,3 +71,65 @@ export const createApiKey = async (
 
   return result[0]
 }
+
+export const deleteApiKey = async (db: Database, apiKeyId: string) => {
+  const currentTime = new Date()
+
+  await db
+    .update(apiKeys)
+    .set({
+      deletedAt: currentTime,
+      updatedAt: currentTime,
+    })
+    .where(eq(apiKeys.id, apiKeyId))
+}
+
+export const updateApiKeyState = async (
+  db: Database,
+  apiKeyId: string,
+  newState: ApiKeyState,
+) => {
+  const currentTime = new Date()
+
+  const updatedApiKeys = await db
+    .update(apiKeys)
+    .set({
+      state: newState,
+      updatedAt: currentTime,
+      stateUpdatedAt: currentTime,
+    })
+    .where(and(eq(apiKeys.id, apiKeyId), isNull(apiKeys.deletedAt)))
+    .returning()
+
+  if (updatedApiKeys.length === 0) {
+    return null
+  }
+
+  // guaranteed to only be 1 max because of the primaryKey filter
+  return updatedApiKeys[0]
+}
+
+const MAX_FETCH_LIMIT = 1000
```

Comment:
this could honestly be even lower like 100 or something even at insane scale we would never hit 1000 for an entity

Reply:
will update!